src/main/java/org/sefglobal/scholarx/service/ProgramService.java

```diff
@@ -116,6 +120,44 @@ public Program updateProgram(long id, Program program) throws ResourceNotFoundEx
      */
     public Program updateState(long id) throws ResourceNotFoundException {
         Optional<Program> program = programRepository.findById(id);
+        List<Mentor> mentors = mentorRepository.findAllByProgramId(id);
+        List<Mentor> approvedMentors = mentorRepository.findAllByProgramIdAndStateIn(id, Collections.singletonList(enrolmentState.APPROVED));
```

Comment:
You can do this without using a collection.
```suggestion
        List<Mentor> approvedMentors = mentorRepository.findAllByProgramIdAndState(id, enrolmentState.APPROVED);
```

Just create a method in mentor repository.
```
List<Mentor> findAllByProgramIdAndState(long id, EnrolmentState states);
```

Reply:
Do this change to line 123 also